File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        if  k == len(nums):
            return sum(nums)/k
        maxNum = float('-inf')
        count = 0

        left = 0
        right = 0

        while right < len(nums):
            count += nums[right]
            if (right-left+1) == k:
                maxNum = max(maxNum,count)
                count -= nums[left]
                left += 1
            right +=1
        return maxNum/k
        

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function(nums, k) {
    let maxAverage = -Infinity
    let start = 0
    let windowSum = 0
    for(let end=0; end < nums.length; end++){
        windowSum += nums[end]
        if(end >= k-1){
            maxAverage= Math.max(maxAverage,windowSum/k)
            windowSum -= nums[start]
            start++
        }

    }
    return maxAverage
};